test/functional/Valhalla/src/org/openj9/test/lworld/ValueTypeUnsafeTests.java

```diff
@@ -88,4 +98,67 @@ static public void testPassingNullToIsFlattenedThrowsException() throws Throwabl
 			myUnsafe.isFlattened(null);
 		});
 	}
+
+	@Test
+	static public void testValueHeaderSizeOfValueType() throws Throwable {
+		long headerSize = myUnsafe.valueHeaderSize(ValueTypePoint2D.class);
+		if (isCompressedRefsEnabled) {
+			assertEquals(headerSize, 4);
+		} else {
+			assertEquals(headerSize, 8);
+		}
+	}
+
+	@Test
+	static public void testValueHeaderSizeOfPaddedValueType() throws Throwable {
+		long headerSize = myUnsafe.valueHeaderSize(ValueTypeWithLongField.class);
+		if (isCompressedRefsEnabled && !isFlatteningEnabled) {
+			assertEquals(headerSize, 4);
+		} else {
+			// the padded size is included in the header size, which will add an extra 4 bytes even when compressed refs are enabled
```

Comment:
> even when compressed refs are enabled

It is added in compressed refs mode and the class has 64 bit field(s) only.

Reply:
I meant to say that despite it being compressed, the header size will be the same as the uncompressed header size because of padding
I'll try to explain that better

Reply:
addressed in d26e081